### DISCUSSION TITLE
$mod+Tab and “auto back and forth”

### DISCUSSION THREAD

**Discussion by p3k** [regolith-linux/regolith-desktop#959] [Q&A]
I could swear `$mod+Tab` switched between two workspaces back and forth if pressed repeatedly in a quick manner, and only switched to the next workspace if consecutively pressed in a slower manner.

This got lost at some time, most likely during a standard `apt upgrade` procedure. At least I cannot recall reconfiguring i3 or regolith settings during the time.

The key pattern is still in my “muscle memory”.

Is it possible to make this work again? I have `workspace_auto_back_and_forth yes` configured.

**Comment by vstath** [comment]
Yes please. I just installed the latest version and I'm missing "back and forth" functionality *badly*

In the very old versions I could do something like

bindsym $mod+b workspace back_and_forth

Is there a new way to do the same? Anyone knows?

**Comment by vstath** [comment]
ok, to answer my own question:

as my user I did:

1) edited file

```
~/.config/regolith3/Xresources
```

and added 

```
workspace_auto_back_and_forth yes
```

2) then edited 

```
~/.config/regolith3/sway/config.d/55_session_keybindings
```

and added

```
bindsym $mod+x workspace back_and_forth
```

Then restarted my session pressing ```<WIN_KEY>+SHIFT+r``` and it just works fine.

Hope it works for you, too

**Reply by p3k** [reply]
thanks for your reply, @vstath – while it works for me, too, using `$mod+x`, it does not with the key combo I desire `$mod+Tab`:

```Xresources
bindsym $mod+Tab workspace back_and_forth
```

```plain
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "workspace back_and_forth"
ERROR: FYI: You are using i3 version 4.21 (2022-09-21)
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "workspace back_and_forth"
ERROR: FYI: You are using i3 version 4.21 (2022-09-21)
ERROR: Could not translate string to key symbol: "-"
ERROR: Could not translate string to key symbol: "scratchpad"
ERROR: Received ConfigureNotify for unknown window 0x03600000
ERROR: Received ConfigureNotify for unknown window 0x03600000
ERROR: Received ConfigureNotify for unknown window 0x03600008
ERROR: Received ExposeEvent for unknown window 0x03600008
ERROR: Received ConfigureNotify for unknown window 0x03600008
ERROR: Received ConfigureNotify for unknown window 0x0360000b
ERROR: Received ExposeEvent for unknown window 0x0360000b
ERROR: Received ConfigureNotify for unknown window 0x0360000b
ERROR: Received ConfigureNotify for unknown window 0x0360000e
ERROR: Received ConfigureNotify for unknown window 0x0360000e
ERROR: Received ConfigureNotify for unknown window 0x0360000e
ERROR: Received ConfigureNotify for unknown window 0x03600011
ERROR: Received ConfigureNotify for unknown window 0x03600011
ERROR: Received ConfigureNotify for unknown window 0x03600011
ERROR: Received ConfigureNotify for unknown window 0x03600014
ERROR: Received ExposeEvent for unknown window 0x03600014
ERROR: Received ConfigureNotify for unknown window 0x03600014
ERROR: Received ConfigureNotify for unknown window 0x03600017
ERROR: Received ExposeEvent for unknown window 0x03600017
ERROR: Received ConfigureNotify for unknown window 0x03600017
ERROR: Received ConfigureNotify for unknown window 0x0360001a
ERROR: Received ExposeEvent for unknown window 0x0360001a
ERROR: Received ConfigureNotify for unknown window 0x0360001a
ERROR: Received ConfigureNotify for unknown window 0x0360001d
ERROR: Received ConfigureNotify for unknown window 0x0360001d
ERROR: Received ConfigureNotify for unknown window 0x0360001d
ERROR: Received ConfigureNotify for unknown window 0x03600020
ERROR: Received ConfigureNotify for unknown window 0x03600020
ERROR: Received ConfigureNotify for unknown window 0x03600020
```

**Reply by p3k** [reply]
I cannot find any duplicate definition with Tab at locations I can think of:

```shell
❯ grep -r Tab ~/.config/regolith2/
# This is the one I added
/home/tobi/.config/regolith2/i3/config.d/99_custom:bindsym $mod+Tab workspace back_and_forth
❯ grep -r Tab /etc/regolith/
# nothing
```

**Reply by garte** [reply]
`wm.binding.ws_next` is your Meta+Tab key combo that would clash with your own Meta+Tab binding. So if you reassign it to something else you won't have any error messages.

In short: Add `wm.binding.ws_next: Mod1+Shift+Mod4+Tab` to your Xresources (or another unused key combo) to overwrite the default Meta+Tab assignment.

**Reply by p3k** [reply]
Yes, I get that. But I see that I want something differently: not a key combo that only switches between two windows back and forth, but one that does the following (as already outlined in this PR’s description):

1. If pressed consecutively in a quick manner: back and forth
2. If pressed consecutively in a slow manner: next window

It might be just me phantasizing this worked before (I am getting older) but still the muscle memory is there :ghost:

**Reply by garte** [reply]
Ah sorry. I've never used Meta-Tab that way, I couldn't remember it ever working like that.
